server/src/main/java/com/descope/java_sample_app/JavaSampleAppApplication.java

```diff
@@ -60,4 +74,35 @@ public ResponseEntity<String> getSecretMessage(HttpServletRequest request) {
 			return new ResponseEntity<>("Error getting authorization header", HttpStatus.UNAUTHORIZED);
 		}
 	}
-}
\ No newline at end of file
+
+	@GetMapping("/start_sso")
+	public ResponseEntity<String> startSSOEndpoint(
+			@RequestParam("tenantId") String tenantId, 
+			@RequestParam(value = "redirectUrl", required = false) String redirectUrl, 
+			@RequestParam(value = "prompt", required = false) String prompt,
+			@RequestParam(value = "loginOptions", required = false) LoginOptions loginOptions) {
+		try {
+			String url = descopeClient.getAuthenticationServices().getSsoServiceProvider().start(tenantId, redirectUrl, prompt,
+			loginOptions);
+			String jsonResponse = "{\"url\": \"" + url + "\"}";
+
+			return ResponseEntity.ok()
+					.contentType(org.springframework.http.MediaType.APPLICATION_JSON)
+					.body(jsonResponse);
+		} catch (DescopeException e) {
+			return new ResponseEntity<>(e.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+	}
+
+	@GetMapping("/authorization-code/callback")
+	public ResponseEntity<String> startSSOEndpoint(
```

Comment:
Here, we set the redirect url to go directly the server endpoint 

This will make it hard to pass the session to the client 

What do you think about setting a ui route that will take the code from the uri, then call this server route that will return the auth info ?

Reply:
For sure! Adjusted the redirect to go to client side which calls backend for token exchange. Adjusted screenshot in readme as well.